### ISSUE TITLE
Notify users outside the App when there are new conversation updates

### ISSUE THREAD

**Issue by camelPhonso** [enBloc-org/kindly#250]
# Expected Behaviour

Users should be prompted to visit Kindly when there are updates to their conversations.

### Spike

We currently have a notification system via emails that notifies users when a new conversation is started. However, extending this to notify users of unread messages could be too costly and has a large potential to drive bad user experience due to their inbox being flooded. As Kindly is not a mobile app we don't have the option to use push notifications. 

# Current Behaviour

Users are notified via email when someone else starts a conversation with them but no other actions trigger a notification beyond in-page displays.

## Steps to Reproduce

If when logged into Kindly you start a new conversation you can find the email notification in your inbox or in the `Inbucket URL` link displayed when you enter `supabase status` onto your terminal.

**Comment by OlenaReukova** [comment]
In our application, we use Nodemailer for sending emails when a user starts a conversation. I am currently testing this functionality using Inbucket with Supabase to validate that emails are sent correctly as part of our app.

However, I'm experiencing an issue during testing. When I try to send a message as a refugee, I don't receive an email notification as a donor in the donor **Inbucket** inbox. What do you think could be causing Supabase Inbucket not to display incoming mail?

**Actual result:** The Inbucket donor inbox is empty. 
**Expected result:** Email notification from trafalgargirls@gmail.com with the subject "New Kindly Message".

Steps to reproduce:

Test on localhost:3000.

1. Log in as the refugee account: refugee+test.reshetniak@gmail.com.
2. Search for an item.
3. Select an item.
4. Click the "Request Item" button.
5. Log out of the refugee profile.
6. Log in as the donor account: donor+test.reshetniak@gmail.com.
7. Check that a message from the refugee has been received.
8. Go to Inbucket.
9. Check the email for donor+test.reshetniak@gmail.com.

**Comment by idellien** [comment]
@OlenaReukova Hi Olena!

I did some investigation and found that the issue with the avatar image is because it’s missing from Supabase storage. This is why the image isn’t loading (you can confirm this in the Supabase Studio: http://localhost:54323/).

While I’m not certain how this affects email notifications, a quick way to unblock this error is by updating the user avatar URL in your local database. Here’s how:

1. .Go to the table editor: http://localhost:54323/project/default/editor
2. Select the profiles table.
3. In the avatar column, update the URL with an existing image. I used this one, which worked for me:
https://undfcbmldjkujposixvn.supabase.co/storage/v1/object/public/images/test_images/toy.jpg?t=2024-07-08T12%3A48%3A39.092Z
 ![Image](https://github.com/user-attachments/assets/9222d709-a35a-4164-bd16-1571508acf05)

After these updates, the error should be resolved! ![Image](https://github.com/user-attachments/assets/9889037d-08ef-4a83-83bc-91c4169b02ae)

Hopefully, this helps you get unblocked. 😊

@camelPhonso  It looks like the image and bucket weren’t set up initially. Do you want  me to create a task for a script to populate these items automatically in the future? Or perhaps I am missing something

**Comment by OlenaReukova** [comment]
@idellien Thank you for investigating this issue. You are correct; the image is absent in Supabase. To fix this error according to our documentation [BEFORE_YOUR_FIRST_ISSUE](https://github.com/enBloc-org/kindly/blob/dev/.github/BEFORE_YOUR_FIRST_ISSUE.md) you need to set up the Bucket manually:
```
Setting Up Supabase Bucket
To store and serve images, Kindly uses a bucket in Supabase. You need to create and configure this bucket manually:

Open the 'Studio URL' in your web browser
Navigate to the "Storage" section
Create a new bucket named 'images'
Set the bucket to be public to ensure it can be accessed without authentication
💡 Note: The bucket and its configuration will be lost if you reset the database or destroy the container. Make sure to create the bucket once again if you reset your environment.
```
I have a status code 200, which indicates that the email was sent.
My question is: how can I test Nodemailer? I tried using Supabase Inbucket, but it did not capture the email.

**Comment by camelPhonso** [comment]
> @camelPhonso It looks like the image and bucket weren’t set up initially. Do you want me to create a task for a script to populate these items automatically in the future? Or perhaps I am missing something

Yes, that would actually be great. I think this is a combination of two things that happened before:

- the staging set up of our buckets is not reproduced in the containerised `dev` environment, so it expects developers to set up a bucket manually like Olena mentioned.

- the seed data on our local does not contain images for profiles. What we see on the page when we run `dev` is the fallback default image the components will render 'onError', but probably setting up actual profile images would be good for future testing scenarios.

**Comment by idellien** [comment]
@OlenaReukova Yeah I did that and even created the image with same name but for some reason it did not worked for me. 

It appears that currently, the “Request Item” button doesn’t send an email notification as part of its functionality within the codebase. Instead, email notifications are handled through the send-email route, which you can find here:
[send-email route](https://github.com/enBloc-org/kindly/blob/dev/app/api/send-email/route.js)

This route is triggered exclusively by the “Request Item” button:
[NewConversationButton.tsx](https://github.com/enBloc-org/kindly/blob/dev/components/buttons/NewConversationButton.tsx)

Additionally, there’s an EnquireButton component that performs a similar function, but it currently isn’t being used anywhere in the project:
[EnquireButton.tsx](https://github.com/enBloc-org/kindly/blob/dev/components/buttons/EnquireButton.tsx)

if you change the step from _“Click the ‘Request Item’ button”_ to _“Click the ‘Message’ button”_, you’ll notice in the logs that the email is attempted but results in a 500 error. This issue occurs because the Gmail app’s username and password are not specified.
<img width="377" alt="image" src="https://github.com/user-attachments/assets/453de1e1-9f45-4015-a39b-8008d96b7ca8">

I have tried to use _trafalgargirls@gmail.com_ but got error "Invalid login: 535-5.7.8 Username and Password not accepted."
Maybe you know correct credentials?
<img width="461" alt="image" src="https://github.com/user-attachments/assets/bbb8e697-a159-42e5-9223-3476ba884105">

**Comment by OlenaReukova** [comment]
@nichgalzin could you check [trafalgargirls@gmail.com](mailto:trafalgargirls@gmail.com), it looks like the password is not valid, please.
